File: lab/python/pares/primeiros_n_pares_impares.py
def primeiros_impares(n):
    cont = 1

    impares = 1
    lista_impares = []

    while cont <= n:
        lista_impares.append(impares)
        impares += 2
        cont += 1

    return lista_impares


def primeiros_pares(n):
    cont = 1

    pares = 2
    lista_pares = []

    while cont <= n:
        lista_pares.append(pares)
        pares += 2
        cont += 1

    return lista_pares


def primeiros(n):
    if not isinstance(n, int):
        raise ValueError("n deve ser do tipo int")

    n = abs(n)

    print("Pares:", " ".join(map(str, primeiros_pares(n))))
    print("Ímpares:", " ".join(map(str, primeiros_impares(n))))


primeiros(4)

File: lab/javascript/pares/primeiros_n_pares_impares.js
function primeiros_impares(n) {
  let cont = 1;
  let impares = 1;
  const lista_impares = [];

  while (cont <= n) {
    lista_impares.push(impares);
    impares += 2;
    cont++;
  }

  return lista_impares;
}

function primeiros_pares(n) {
  let cont = 1;
  let pares = 2;
  const lista_pares = [];

  while (cont <= n) {
    lista_pares.push(pares);
    pares += 2;
    cont++;
  }

  return lista_pares;
}

function primeiros(n) {
  if (!Number.isInteger(n)) {
    throw new Error("n deve ser do tipo int");
  }

  n = Math.abs(n);

  console.log("Pares:", primeiros_pares(n).join(" "));
  console.log("Ímpares:", primeiros_impares(n).join(" "));
}

// -------------------- Teste --------------------
primeiros(4);
